akka-distributed-data/src/test/scala/akka/cluster/ddata/ORMapSpec.scala

```diff
@@ -110,6 +141,280 @@ class ORMapSpec extends WordSpec with Matchers {
+
+      val merged1 = (m1 merge m2d) mergeDelta m2u.delta.get
+
+      merged1.entries("a").elements should be(Set("A"))
+      // note that B1 is lost as it was added and removed earlier in timeline than B2
+      merged1.entries("b").elements should be(Set("B2", "B3", "B4"))
+    }
+
+    "not have anomalies for remove+updated scenario and deltas 8" in {
+      val m1 = ORMap.empty.put(node1, "a", GSet.empty + "A")
+        .put(node1, "b", GSet.empty + "B").put(node2, "b", GSet.empty + "B")
+      val m2 = ORMap.empty.put(node2, "c", GSet.empty + "C")
+
+      val merged1 = m1 merge m2
+
+      val m3 = merged1.resetDelta.remove(node1, "b").remove(node2, "b")
+      val m4 = merged1.resetDelta.put(node2, "b", GSet.empty + "B2").put(node2, "b", GSet.empty + "B3")
+
+      val merged2 = m3 merge m4
+
+      merged2.entries("a").elements should be(Set("A"))
+      merged2.entries("b").elements should be(Set("B3"))
+      merged2.entries("c").elements should be(Set("C"))
+
+      val merged3 = (merged1 mergeDelta m3.delta.get) mergeDelta m4.delta.get
+
+      merged3.entries("a").elements should be(Set("A"))
+      merged3.entries("b").elements should be(Set("B3"))
+      merged3.entries("c").elements should be(Set("C"))
+    }
+
+    "not have anomalies for remove+updated scenario and deltas 9" in {
+      val m1 = ORMap.empty.put(node1, "a", GSet.empty + "A")
+        .put(node1, "b", GSet.empty + "B").put(node2, "b", GSet.empty + "B")
+      val m2 = ORMap.empty.put(node2, "c", GSet.empty + "C")
+
+      val merged1 = m1 merge m2
+
+      val m3 = merged1.resetDelta.remove(node1, "b").remove(node2, "b")
+      val m4 = merged1.resetDelta.updated(node2, "b", GSet.empty[String])(_.add("B2"))
+        .updated(node2, "b", GSet.empty[String])(_.add("B3"))
+
+      val merged2 = m3 merge m4
+
+      merged2.entries("a").elements should be(Set("A"))
+      merged2.entries("b").elements should be(Set("B2", "B3"))
+      merged2.entries("c").elements should be(Set("C"))
+
+      val merged3 = (merged1 mergeDelta m3.delta.get) mergeDelta m4.delta.get
+
+      merged3.entries("a").elements should be(Set("A"))
+      merged3.entries("b").elements should be(Set("B2", "B3"))
+      merged3.entries("c").elements should be(Set("C"))
+    }
+
+    "not have anomalies for remove+updated scenario and deltas 10" in {
+      val m1 = ORMap.empty.put(node2, "a", GSet.empty + "A")
+        .put(node2, "b", GSet.empty + "B")
+
+      val m3 = m1.resetDelta.remove(node2, "b")
+      val m4 = m3.resetDelta.put(node2, "b", GSet.empty + "B2").updated(node2, "b", GSet.empty[String])(_.add("B3"))
+
+      val merged2 = m3 merge m4
+
+      merged2.entries("a").elements should be(Set("A"))
+      merged2.entries("b").elements should be(Set("B2", "B3"))
+
+      val merged3 = m3 mergeDelta m4.delta.get
+
+      merged3.entries("a").elements should be(Set("A"))
+      merged3.entries("b").elements should be(Set("B2", "B3"))
```

Comment:
Still works without tombstones, because now full value is being propagated, not its delta